src/server/gc_worker.rs

```diff
@@ -1059,12 +1058,49 @@ pub struct GCWorker<E: Engine> {
 
     ratio_threshold: f64,
 
+    /// How many strong references. The worker will be stopped
+    /// once there are no more references.
+    refs: Arc<atomic::AtomicUsize>,
     worker: Arc<Mutex<Worker<GCTask>>>,
     worker_scheduler: worker::Scheduler<GCTask>,
 
     gc_manager_handle: Arc<Mutex<Option<GCManagerHandle>>>,
 }
 
+impl<E: Engine> Clone for GCWorker<E> {
```

Comment:
Now the `GCWoker` starts explicitly but stops implicitly. It's weird to me. Maybe we can call `GCWorker.stop()` in `run_raft_server()` like other workers.

Reply:
how about creating a `GCWorkerBuilder` and call `start` in its `build` function ? So, `GCWoker` can be started and stoped implicitly .